lib/Models/Terria.js

```diff
@@ -550,17 +550,25 @@ function generateInitializationUrl(url) {
 
 function loadInitSources(terria, initSources) {
     return when.all(initSources.map(loadInitSource.bind(undefined, terria)), function(initSources) {
-        var promises = [];
+        var promises = [], updateSources = [];
 
         for (var i = 0; i < initSources.length; ++i) {
             var initSource = initSources[i];
             if (!defined(initSource)) {
                 continue;
             }
-            promises.push(terria.addInitSource(initSource));
+
+            if (initSource.catalogOnlyUpdatesExistingItems) {
```

Comment:
Sure, if that's the intention then it make sense to do it that way. I was worried about causing previously concurrent data loads to now happen one after the other.